src/doctor.ts

```diff
@@ -1,13 +1,22 @@
-export interface DoctorResult {
+type NonEmptyArray<A> = { 0: A } & A[];
+
```

Comment:
Handmade `NonEmptyArray` definition, represented as an intersection type between a normal array (tail) and the 0th element (head)